client/components/MovieGrid.js

```diff
@@ -1,5 +1,5 @@
-import React, { useState, useEffect } from 'react';
-import { getTopMovies } from '../api/MovieApi';
+import React, { useState, useEffect, forwardRef, useImperativeHandle } from 'react';
```

Comment:
Interesting! Never seen useImperativeHandle before, but can see how it would be useful to force-grab refs. Generally I think there's probably some way to do this more gracefully, but TIL, a very interesting tool to know about